fuzzing/private/java.bzl

```diff
@@ -0,0 +1,168 @@
+            segment = path_segments[i]
+            if "src" == segment or (is_src and ("javatests" == segment or "java" == segment)):
+                next = path_segments[i + 1]
+                if ("com" == next or "org" == next or "net" == next):
+                    root_index = i
+                elif "src" == segment:
+                    check_maven_index = i
+                break
+
+    if check_maven_index >= 0 and check_maven_index + 2 < len(path_segments):
+        next = path_segments[check_maven_index + 1]
+        if "main" == next or "test" == next:
+            next = path_segments[check_maven_index + 2]
+            if "java" == next or "resources" == next:
+                root_index = check_maven_index + 2
+
+    return root_index
+
+def _jazzer_fuzz_binary_script(ctx):
+    script = ctx.actions.declare_file(ctx.label.name)
+
+    script_template = """
+exec "{driver}" \
+    --agent_path="{agent}" \
+    --cp="{deploy_jar}" \
+    --jvm_args="-Djava.library.path={library_path}" \
+    "$@"
+"""
+
+    # Collect the directories that contain the native library dependencies.
+    # TODO: Improve once
+    #  https://github.com/bazelbuild/bazel/commit/381a519dfc082d4c62096c4ce77ead1c2e0410d8
+    #  has been released.
+    native_files = [native_dep[DefaultInfo].files for native_dep in ctx.attr.native_deps]
+    native_paths = [file.short_path for file in depset(transitive = native_files).to_list()]
+    native_dirs = [path[:path.rfind("/")] for path in native_paths]
+
+    script_content = script_template.format(
+        driver = ctx.executable.driver.short_path,
+        agent = ctx.file._agent.short_path,
+        deploy_jar = ctx.file.target_deploy_jar.short_path,
+        library_path = ":".join(native_dirs),
+    )
+    ctx.actions.write(script, script_content, is_executable = True)
+    return script
+
+def _jazzer_fuzz_binary_impl(ctx):
+    script = _jazzer_fuzz_binary_script(ctx)
+
+    runfiles = ctx.runfiles()
+    runfiles = runfiles.merge(ctx.attr.driver[DefaultInfo].default_runfiles)
+    runfiles = runfiles.merge(ctx.runfiles([ctx.file._agent]))
+    runfiles = runfiles.merge(ctx.runfiles([ctx.file.target_deploy_jar]))
+    for native_dep in ctx.attr.native_deps:
+        runfiles = runfiles.merge(native_dep[DefaultInfo].default_runfiles)
+    return [DefaultInfo(executable = script, runfiles = runfiles)]
+
+jazzer_fuzz_binary = rule(
+    implementation = _jazzer_fuzz_binary_impl,
+    doc = """
+Rule that creates a binary that invokes Jazzer on the specified target.
+""",
+    attrs = {
+        "driver": attr.label(
+            default = Label("@jazzer//driver:jazzer_driver"),
+            doc = "The Jazzer driver binary used to fuzz the target.",
+            executable = True,
+            # Build in target configuration rather than host because the driver
+            # uses transitions to set the correct C++ standard for its
+            # dependencies.
+            cfg = "target",
```

Comment:
The `jazzer_driver` target uses a transition to apply the copt`-std=c++17` to it and all dependencies. Without this, it seems that any workspace depending on `rules_fuzzing` would need to set this in their `.bazelrc`. If you know a better way to build the driver with C++17 without affecting the rest of the workspace, please let me know.